pkgs/servers/x11/xorg/overrides.nix

```diff
@@ -134,171 +752,146 @@ self: super:
+    ];
     meta = attrs.meta // {
-      mainProgram = "xdpyinfo";
+      license = with lib.licenses; [
+        hpndSellVariant
+        mit
+      ];
     };
   });
 
-  xdm = super.xdm.overrideAttrs (attrs: {
-    buildInputs = attrs.buildInputs ++ [ libxcrypt ];
-    configureFlags = attrs.configureFlags or [] ++ [
-      "ac_cv_path_RAWCPP=${stdenv.cc.targetPrefix}cpp"
-    ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform)
-      # checking for /dev/urandom... configure: error: cannot check for file existence when cross compiling
-      [ "ac_cv_file__dev_urandom=true" "ac_cv_file__dev_random=true" ];
-    meta = attrs.meta // { mainProgram = "xdm"; };
-  });
-
-  # Propagate some build inputs because of header file dependencies.
-  # Note: most of these are in Requires.private, so maybe builder.sh
-  # should propagate them automatically.
-  libXt = super.libXt.overrideAttrs (attrs: {
-    preConfigure = ''
-      sed 's,^as_dummy.*,as_dummy="\$PATH",' -i configure
-    '';
-    configureFlags = attrs.configureFlags or []
-      ++ malloc0ReturnsNullCrossFlag;
-    propagatedBuildInputs = attrs.propagatedBuildInputs or [] ++ [ xorg.libSM ];
-    depsBuildBuild = [ buildPackages.stdenv.cc ];
-    CPP = if stdenv.isDarwin then "clang -E -" else "${stdenv.cc.targetPrefix}cc -E -";
-    outputDoc = "devdoc";
-    outputs = [ "out" "dev" "devdoc" ];
-  });
-
-  luit = super.luit.overrideAttrs (attrs: {
-    # See https://bugs.freedesktop.org/show_bug.cgi?id=47792
-    # Once the bug is fixed upstream, this can be removed.
-    configureFlags = [ "--disable-selective-werror" ];
-
-    buildInputs = attrs.buildInputs ++ [libiconv];
-    meta = attrs.meta // { mainProgram = "luit"; };
-  });
-
-  libICE = super.libICE.overrideAttrs (attrs: {
-    outputs = [ "out" "dev" "doc" ];
-  });
-
-  libXcomposite = super.libXcomposite.overrideAttrs (attrs: {
-    outputs = [ "out" "dev" ];
-    propagatedBuildInputs = attrs.propagatedBuildInputs or [] ++ [ xorg.libXfixes ];
-  });
-
-  libXaw = super.libXaw.overrideAttrs (attrs: {
-    outputs = [ "out" "dev" "devdoc" ];
-    propagatedBuildInputs = attrs.propagatedBuildInputs or [] ++ [ xorg.libXmu ];
-  });
-
-  libXcursor = super.libXcursor.overrideAttrs (attrs: {
-    outputs = [ "out" "dev" ];
+  libXfont = prev.libXfont.overrideAttrs (attrs: {
+    outputs = [
+      "out"
+      "dev"
+    ];
+    propagatedBuildInputs = attrs.propagatedBuildInputs or [ ] ++ [ freetype ]; # propagate link reqs. like bzip2
+    # prevents "misaligned_stack_error_entering_dyld_stub_binder"
+    configureFlags = lib.optional isDarwin "CFLAGS=-O0";
+    meta = attrs.meta // {
+      license = with lib.licenses; [
+        mit
+        mitOpenGroup
+        smlnj
+        hpndSellVariant
+        bsdOriginalUC
+        bsd2
+        # lib/font/fontfile/gunzip.c
+        # > intended for inclusion in X11 public releases.
+        # as well as the last license in COPYING grants no freedoms
+        unfree
```

Comment:
In the same way that submitting code to Nixpkgs intended for inclusion in its public releases constitutes obvious intent to license your code under the terms those public releases are released under, I’m quite sure the same applies here.

Reply:
i see your point, however i would interpret
> intended for inclusion in X11 public releases.
as "exclusive license to redistribute in X11 public releases"
are xorg packages in nixpkgs public releases of X11? maybe

Reply:
I think that interpretation is reading too much into the notice, but perhaps it’d be a good idea to report this upstream as well? It’s at least more ambiguous than the Fujitsu stuff.